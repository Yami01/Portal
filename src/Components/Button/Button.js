/**
 *
 * Button
 * @format
 * @flow
 *
 */

import React, { memo, useCallback, useMemo } from 'react'
import {
  Animated,
  TouchableWithoutFeedback,
  useWindowDimensions,
  View,
} from 'react-native'
import {
  ActivityIndicator,
  TouchableRipple,
  useTheme,
} from 'react-native-paper'
import Color from 'color'
import type { AbstractComponent, Node } from 'React'
import type {
  TextStyleProp,
  ViewStyleProp,
} from 'react-native/Libraries/StyleSheet/StyleSheet'

import {
  BOTTOM_ICON_COLOR,
  BOTTOM_ICON_CONTAINED_COLOR,
  BUTTON_BORDER_RADIUS,
  BUTTON_COLOR,
  BUTTON_ICON_GROUP,
  BUTTON_ICON_MEDIUM_SIZE,
  BUTTON_ICON_SIZE,
  BUTTON_ICON_SMALL_SIZE,
  BUTTON_NUMBER_OF_LINES,
  BUTTON_OUTLINED_BACKGROUND_COLOR,
  LARGE_BUTTON_HEIGHT,
  MEDIUM_BUTTON_HEIGHT,
  SMALL_BUTTON_HEIGHT,
} from './config'

import ButtonStyles from './styles'
import type { IconPropsType, PropsType } from './types'
import { Colors } from '@/Theme/Variables'
import Text from '@/Components/Text'
import Icons from '@/Assets/Icons'

const Button: AbstractComponent<PropsType> = memo((props: PropsType): Node => {
  const {
    size = 'large',
    alignment,
    borderRadius = BUTTON_BORDER_RADIUS,
    children,
    color = BUTTON_COLOR,
    compact,
    contentStyle,
    disabled: propsDisabled = false,
    halfWidth = false,
    autoWidth = false,
    iconSize,
    labelStyle,
    leftIcon,
    loading,
    onPress,
    rightIcon,
    style: propsStyle,
    uppercase,
    underline,
    variation = 'contained',
    testID,
    textTestID,
    maxFontScale,
    numberOfLines = BUTTON_NUMBER_OF_LINES,
  } = props

  const { animation, fonts } = useTheme()

  const disabled = useMemo((): boolean => {
    return loading || !!propsDisabled
  }, [loading, propsDisabled])

  const { current: elevation } = React.useRef(
    new Animated.Value(variation === 'contained' ? 2 : 0),
  )

  const disableColor = useCallback((originalColor: string): string => {
    return Color(originalColor).lighten(0.75).desaturate(0.3).rgb().string()
  }, [])

  const backgroundColor = useMemo((): string => {
    if (variation === 'contained') {
      return color
    }
    if (variation === 'outlined') {
      return BUTTON_OUTLINED_BACKGROUND_COLOR
    }
    return 'transparent'
  }, [color, variation])

  const disabledBackgroundColor = useMemo((): string => {
    if (variation === 'contained') {
      return disableColor(backgroundColor)
    }
    return backgroundColor
  }, [variation, backgroundColor, disableColor])

  const borderColor = useMemo((): string => {
    if (variation === 'outlined') {
      return disabled ? disableColor(color) : color
    }
    return 'transparent'
  }, [variation, disabled, disableColor, color])

  const borderWidth = useMemo((): number => {
    return variation === 'outlined' ? 1 : 0
  }, [variation])

  const textColor = useMemo((): string => {
    if (variation === 'contained') {
      return Colors.white
    }
    if (disabled) {
      return disableColor(color)
    }
    return color
  }, [variation, disabled, color, disableColor])

  const rippleColor = useMemo(
    (): string => Color(textColor).alpha(0.32).rgb().string(),
    [textColor],
  )

  const buttonStyle = useMemo((): ViewStyleProp => {
    return {
      backgroundColor: disabled ? disabledBackgroundColor : backgroundColor,
      borderColor,
      borderWidth,
      borderRadius,
    }
  }, [
    backgroundColor,
    borderColor,
    borderRadius,
    borderWidth,
    disabled,
    disabledBackgroundColor,
  ])

  const touchableStyle = useMemo((): ViewStyleProp => {
    return {
      borderRadius,
    }
  }, [borderRadius])

  const finalLabelStyle = useMemo((): TextStyleProp => {
    const isTextStyle = variation === 'text'
    const defaultLabelStyle = isTextStyle
      ? ButtonStyles.textStyleLabel
      : [ButtonStyles.label, fonts.semiBold]
    let labelSizeStyle = isTextStyle
      ? ButtonStyles.largeTextStyleLabel
      : ButtonStyles.largeLabel
    if (size === 'small') {
      labelSizeStyle = isTextStyle
        ? ButtonStyles.smallTextStyleLabel
        : ButtonStyles.smallLabel
    } else if (size === 'medium') {
      labelSizeStyle = isTextStyle
        ? ButtonStyles.mediumTextStyleLabel
        : ButtonStyles.mediumLabel
    }
    return [
      defaultLabelStyle,
      labelSizeStyle,
      compact && ButtonStyles.compactLabel,
      uppercase && ButtonStyles.uppercaseLabel,
      underline && ButtonStyles.underlineLabel,
      { color: textColor },
      labelStyle,
    ]
  }, [
    compact,
    labelStyle,
    size,
    textColor,
    underline,
    uppercase,
    variation,
    fonts,
  ])

  const { fontScale } = useWindowDimensions()

  const finalFontScale = useMemo((): number => {
    return maxFontScale ? Math.min(maxFontScale, fontScale) : fontScale
  }, [fontScale, maxFontScale])

  const finalButtonStyle = useMemo((): ViewStyleProp => {
    let buttonSizeStyle = ButtonStyles.largeButton
    let fontScaleStyle = { height: LARGE_BUTTON_HEIGHT * finalFontScale }
    if (size === 'small') {
      buttonSizeStyle = ButtonStyles.smallButton
      fontScaleStyle = { height: SMALL_BUTTON_HEIGHT * finalFontScale }
    } else if (size === 'medium') {
      buttonSizeStyle = ButtonStyles.mediumButton
      fontScaleStyle = { height: MEDIUM_BUTTON_HEIGHT * finalFontScale }
    }
    return [
      ButtonStyles.button,
      buttonSizeStyle,
      fontScaleStyle,
      compact && ButtonStyles.compact,
      buttonStyle,
      propsStyle,
    ]
  }, [finalFontScale, size, compact, buttonStyle, propsStyle])

  const finalIconSize = useMemo((): number => {
    let resultSize = iconSize || BUTTON_ICON_SIZE
    if (!iconSize) {
      if (size === 'small') {
        resultSize = BUTTON_ICON_SMALL_SIZE
      } else if (size === 'medium') {
        resultSize = BUTTON_ICON_MEDIUM_SIZE
      }
    }
    return resultSize * fontScale
  }, [fontScale, iconSize, size])

  const handlePress = useCallback(() => {
    if (onPress) {
      onPress()
    }
  }, [onPress])

  const handlePressIn = useCallback(() => {
    if (variation === 'contained') {
      const { scale } = animation
      Animated.timing(elevation, {
        toValue: 8,
        duration: 200 * scale,
        useNativeDriver: true,
      }).start()
    }
  }, [animation, elevation, variation])

  const handlePressOut = useCallback(() => {
    if (variation === 'contained') {
      const { scale } = animation
      Animated.timing(elevation, {
        toValue: 2,
        duration: 150 * scale,
        useNativeDriver: true,
      }).start()
    }
  }, [animation, elevation, variation])

  const renderIcon = useCallback(
    ({
      group = BUTTON_ICON_GROUP,
      name,
      iconColor = variation === 'contained'
        ? BOTTOM_ICON_CONTAINED_COLOR
        : BOTTOM_ICON_COLOR,
    }: IconPropsType): Node => {
      if (group && name) {
        const iconNode = Icons[group]?.(
          name,
          {},
          finalIconSize,
          finalIconSize,
          disabled ? disableColor(iconColor) : iconColor,
        )
        return <View style={ButtonStyles.iconContainer}>{iconNode}</View>
      }
      return null
    },
    [disableColor, disabled, finalIconSize, variation],
  )

  const finalContainerStyle = useMemo((): ViewStyleProp => {
    let alignmentStyle
    switch (alignment) {
      case 'left':
        alignmentStyle = {
          alignItems: 'flex-start',
        }
        break
      case 'right':
        alignmentStyle = {
          alignItems: 'flex-end',
        }
        break
      case 'center':
        alignmentStyle = {
          alignItems: 'center',
        }
        break
      default:
        alignmentStyle = {}
    }

    return [
      ButtonStyles.fullWidthContainer,
      halfWidth && ButtonStyles.halfWidthContainer,
      autoWidth && ButtonStyles.autoWidthContainer,
      alignmentStyle,
    ]
  }, [halfWidth, autoWidth, alignment])

  const renderButtonContent = useCallback((): Node => {
    return (
      <View>
        <View
          style={[
            ButtonStyles.content,
            contentStyle,
            loading ? ButtonStyles.contentLoading : {},
          ]}
        >
          {leftIcon && renderIcon(leftIcon)}
          {!!children && (
            <Text
              numberOfLines={numberOfLines}
              style={finalLabelStyle}
              testID={textTestID}
              maxFontSizeMultiplier={maxFontScale}
            >
              {children}
            </Text>
          )}
          {rightIcon && renderIcon(rightIcon)}
        </View>
        {loading ? (
          <View style={ButtonStyles.spinnerContainer}>
            <ActivityIndicator color={textColor} size={finalIconSize} />
          </View>
        ) : null}
      </View>
    )
  }, [
    children,
    contentStyle,
    finalIconSize,
    finalLabelStyle,
    leftIcon,
    loading,
    maxFontScale,
    renderIcon,
    rightIcon,
    textColor,
    textTestID,
    numberOfLines,
  ])

  const renderTouchableButton = useCallback((): Node => {
    const touchableProps = {
      disabled,
      onPress: handlePress,
      onPressIn: handlePressIn,
      onPressOut: handlePressOut,
    }

    if (variation === 'text') {
      return (
        <TouchableWithoutFeedback {...touchableProps}>
          {renderButtonContent()}
        </TouchableWithoutFeedback>
      )
    }
    return (
      <TouchableRipple
        {...touchableProps}
        accessibilityComponentType="button"
        borderless
        delayPressIn={0}
        rippleColor={rippleColor}
        style={touchableStyle}
      >
        {renderButtonContent()}
      </TouchableRipple>
    )
  }, [
    disabled,
    handlePress,
    handlePressIn,
    handlePressOut,
    renderButtonContent,
    rippleColor,
    touchableStyle,
    variation,
  ])

  return (
    <View testID={testID} style={finalContainerStyle}>
      <View style={finalButtonStyle}>{renderTouchableButton()}</View>
    </View>
  )
})

export default Button
