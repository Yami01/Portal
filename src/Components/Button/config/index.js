/**
 *
 * @format
 * @flow
 *
 */
import Colors from "react-native/Libraries/NewAppScreen/components/Colors"

export const LARGE_BUTTON_HEIGHT = 56
export const MEDIUM_BUTTON_HEIGHT = 48
export const SMALL_BUTTON_HEIGHT = 32

export const BUTTON_ICON_SIZE = 20
export const BUTTON_ICON_MEDIUM_SIZE = 18
export const BUTTON_ICON_SMALL_SIZE = 16
export const BUTTON_ICON_GROUP = 'general'
export const BOTTOM_ICON_CONTAINED_COLOR = Colors.white
export const BOTTOM_ICON_COLOR = Colors.error
export const BUTTON_BORDER_RADIUS = 8
export const BUTTON_COLOR = Colors.primary
export const BUTTON_OUTLINED_BACKGROUND_COLOR = Colors.white
export const BUTTON_NUMBER_OF_LINES = 1
