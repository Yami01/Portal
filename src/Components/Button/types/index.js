/**
 *
 * @format
 * @flow
 *
 */

import type { Node } from "react"
import type { TextStyleProp, ViewStyleProp } from "react-native/Libraries/StyleSheet/StyleSheet"

export type IconPropsType = {
  group: string,
  name: string,
  iconColor?: string,
}

export type PropsType = {
  borderRadius?: number,
  children?: Node,
  color?: string,
  compact?: boolean,
  contentStyle?: ViewStyleProp,
  disabled?: boolean,
  halfWidth?: boolean,
  autoWidth?: boolean,
  iconSize?: number,
  leftIcon?: IconPropsType,
  labelStyle?: TextStyleProp,
  loading?: boolean,
  onPress?: () => any,
  rightIcon?: IconPropsType,
  style?: ViewStyleProp,
  uppercase?: boolean,
  underline?: boolean,
  alignment?: 'left' | 'center' | 'right',
  size?: 'small' | 'medium' | 'large',
  variation?: 'text' | 'outlined' | 'contained',
  testID?: string,
  textTestID?: string,
  maxFontScale?: number,
  numberOfLines?: number,
}

export type ButtonStylesType = {
  button: ViewStyleProp,
  smallButton: ViewStyleProp,
  mediumButton: ViewStyleProp,
  largeButton: ViewStyleProp,
  iconContainer: ViewStyleProp,
  compact: ViewStyleProp,
  compactLabel: ViewStyleProp,
  content: ViewStyleProp,
  contentLoading: ViewStyleProp,
  fullWidthContainer: ViewStyleProp,
  halfWidthContainer: ViewStyleProp,
  autoWidthContainer: ViewStyleProp,
  label: TextStyleProp,
  smallLabel: TextStyleProp,
  mediumLabel: TextStyleProp,
  largeLabel: TextStyleProp,
  textStyleLabel: TextStyleProp,
  smallTextStyleLabel: TextStyleProp,
  mediumTextStyleLabel: TextStyleProp,
  largeTextStyleLabel: TextStyleProp,
  spinnerContainer: ViewStyleProp,
  uppercaseLabel: TextStyleProp,
  underlineLabel: TextStyleProp,
}
