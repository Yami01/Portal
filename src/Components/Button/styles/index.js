/**
 *
 * @format
 * @flow
 *
 */

import { StyleSheet } from "react-native"
import { LARGE_BUTTON_HEIGHT, MEDIUM_BUTTON_HEIGHT, SMALL_BUTTON_HEIGHT } from "../config"

import type { ButtonStylesType } from "../types"

const ButtonStyles: ButtonStylesType = StyleSheet.create({
  button: {
    borderStyle: 'solid',
  },
  largeButton: {
    minWidth: LARGE_BUTTON_HEIGHT,
    height: LARGE_BUTTON_HEIGHT,
  },
  mediumButton: {
    minWidth: MEDIUM_BUTTON_HEIGHT,
    height: MEDIUM_BUTTON_HEIGHT,
  },
  smallButton: {
    minWidth: SMALL_BUTTON_HEIGHT,
    height: SMALL_BUTTON_HEIGHT,
  },
  compact: {
    minWidth: 'auto',
    height: 'auto',
  },
  compactLabel: {
    marginHorizontal: 8,
  },
  content: {
    height: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  contentLoading: {
    opacity: 0,
  },
  iconContainer: {
    marginHorizontal: 4,
  },
  textStyleLabel: {
    textAlign: 'center',
    marginRight: 4,
  },
  largeTextStyleLabel: {
    fontSize: 16,
    lineHeight: 24,
  },
  mediumTextStyleLabel: {
    fontSize: 14,
    lineHeight: 20,
  },
  smallTextStyleLabel: {
    fontSize: 12,
    lineHeight: 18,
  },
  label: {
    textAlign: 'center',
    marginHorizontal: 4,
    letterSpacing: 0.1,
  },
  largeLabel: {
    fontSize: 20,
    lineHeight: 24,
  },
  mediumLabel: {
    fontSize: 16,
    lineHeight: 20,
  },
  smallLabel: {
    fontSize: 14,
    lineHeight: 20,
  },
  spinnerContainer: {
    position: 'absolute',
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
  uppercaseLabel: {
    textTransform: 'uppercase',
  },
  underlineLabel: {
    textDecorationLine: 'underline',
  },
  fullWidthContainer: {
    width: '100%',
  },
  halfWidthContainer: {
    width: '45%',
  },
  autoWidthContainer: {
    width: 'auto',
  },
})

export default ButtonStyles
