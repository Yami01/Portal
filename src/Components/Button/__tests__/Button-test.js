/**
 *
 * Button Test
 * @format
 * @flow
 *
 */
import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'
import '@testing-library/jest-dom/extend-expect'
import Button from '../Button'

describe('Button', () => {
  const iconTestId = 'Icon'
  const iconProps = {
    group: 'general',
    name: 'arrow_forward',
  }

  test('Test if button renders properly', () => {
    const tree = renderer
      .create(<Button onPress={() => {}} leftIcon={iconProps} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('is not be clickable when disabled prop is true', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test'

    const { getByText } = render(
      <Button disabled onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const button = getByText(buttonLabel)
    fireEvent.press(button)
    expect(button).toBeDisabled()
    expect(onPressMock).not.toHaveBeenCalled()
  })

  test('is a11y works for testing', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'
    const testId = 'a11yTestButton'

    const { findByTestId } = render(
      <Button variation="text" onPress={onPressMock} testID={testId}>
        {buttonLabel}
      </Button>,
    )

    const button = findByTestId(testId)
    expect(button).not.toBeNull()
  })

  test('is text variation button properly render', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'

    const { getByText } = render(
      <Button variation="text" onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const button = getByText(buttonLabel)
    expect(button).not.toBeNull()
  })

  test('is text variation with left icon button properly render', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'

    const { findAllByTestId } = render(
      <Button variation="text" leftIcon={iconProps} onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const buttonWithIconLeft = findAllByTestId(iconTestId)
    expect(buttonWithIconLeft).not.toBeNull()
    expect(buttonWithIconLeft).not.toBe({})
  })

  test('is text variation with right icon button properly render', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'

    const { findAllByTestId } = render(
      <Button variation="text" rightIcon={iconProps} onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const buttonWithIconRight = findAllByTestId(iconTestId)
    expect(buttonWithIconRight).not.toBeNull()
    expect(buttonWithIconRight).not.toBe({})
  })

  test('is outlined variation with left icon button properly render', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'

    const { findAllByTestId } = render(
      <Button variation="outlined" leftIcon={iconProps} onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const buttonWithIconLeft = findAllByTestId(iconTestId)
    expect(buttonWithIconLeft).not.toBeNull()
    expect(buttonWithIconLeft).not.toBe({})
  })

  test('is outlined variation with right icon button properly render', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'

    const { findAllByTestId } = render(
      <Button variation="outlined" rightIcon={iconProps} onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const buttonWithIconRight = findAllByTestId(iconTestId)
    expect(buttonWithIconRight).not.toBeNull()
    expect(buttonWithIconRight).not.toBe({})
  })

  test('is contained variation with left icon button properly render', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'

    const { findAllByTestId } = render(
      <Button leftIcon={iconProps} onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const buttonWithIconLeft = findAllByTestId(iconTestId)
    expect(buttonWithIconLeft).not.toBeNull()
    expect(buttonWithIconLeft).not.toBe({})
  })

  test('is contained variation with right icon button properly render', () => {
    const onPressMock = jest.fn()
    const buttonLabel = 'Test Text Button'

    const { findAllByTestId } = render(
      <Button rightIcon={iconProps} onPress={onPressMock}>
        {buttonLabel}
      </Button>,
    )

    const buttonWithIconRight = findAllByTestId(iconTestId)
    expect(buttonWithIconRight).not.toBeNull()
    expect(buttonWithIconRight).not.toBe({})
  })

  test('Text should not render if it does not exists', () => {
    const textTestID = 'button_text_test_id'
    const { queryByTestId } = render(
      <Button
        underline
        onPress={() => {}}
        leftIcon={iconProps}
        textTestID={textTestID}
      />,
    )
    expect(queryByTestId(textTestID)).toBeNull()
  })

  test('Test if button renders properly with underline props', () => {
    const textTestID = 'Test Underline'
    const buttonText = 'Try me!'
    const { queryByTestId } = render(
      <Button
        underline
        onPress={() => {}}
        leftIcon={iconProps}
        textTestID={textTestID}
      >
        {buttonText}
      </Button>,
    )

    expect(queryByTestId(textTestID)).toHaveStyle({
      textDecorationLine: 'underline',
    })
  })

  test('Test if button renders properly with uppercase props', () => {
    const textTestID = 'Test Uppercase'
    const buttonText = 'Try me!'
    const { queryByTestId } = render(
      <Button
        uppercase
        onPress={() => {}}
        leftIcon={iconProps}
        textTestID={textTestID}
      >
        {buttonText}
      </Button>,
    )

    expect(queryByTestId(textTestID)).toHaveStyle({
      textTransform: 'uppercase',
    })
  })

  test('Test if button renders properly with compact props', () => {
    const tree = renderer
      .create(<Button compact onPress={() => {}} leftIcon={iconProps} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Test if button renders properly with size small props', () => {
    const tree = renderer
      .create(<Button size="small" onPress={() => {}} leftIcon={iconProps} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Test if button renders properly with size medium props', () => {
    const tree = renderer
      .create(<Button size="medium" onPress={() => {}} leftIcon={iconProps} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Test if button renders properly with text style and size small props', () => {
    const tree = renderer
      .create(
        <Button
          variation="text"
          size="small"
          onPress={() => {}}
          leftIcon={iconProps}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Test if button renders properly with text style and size medium props', () => {
    const tree = renderer
      .create(
        <Button
          variation="text"
          size="medium"
          onPress={() => {}}
          leftIcon={iconProps}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Test if button renders properly with autoHeight props', () => {
    const testID = 'testButton'
    const { queryByTestId } = render(
      <Button
        underline
        onPress={() => {}}
        leftIcon={iconProps}
        testID={testID}
        autoWidth
      />,
    )

    expect(queryByTestId(testID)).toHaveStyle({ width: 'auto' })
  })
})
