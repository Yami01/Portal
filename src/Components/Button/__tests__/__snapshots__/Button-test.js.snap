// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test if button renders properly 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
      false,
      false,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderStyle": "solid",
        },
        Object {
          "height": 56,
          "minWidth": 56,
        },
        Object {
          "height": 112,
        },
        undefined,
        Object {
          "backgroundColor": "#FFF",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 8,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
              },
              undefined,
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginHorizontal": 4,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Button Test if button renders properly with compact props 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
      false,
      false,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderStyle": "solid",
        },
        Object {
          "height": 56,
          "minWidth": 56,
        },
        Object {
          "height": 112,
        },
        Object {
          "height": "auto",
          "minWidth": "auto",
        },
        Object {
          "backgroundColor": "#FFF",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 8,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
              },
              undefined,
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginHorizontal": 4,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Button Test if button renders properly with size medium props 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
      false,
      false,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderStyle": "solid",
        },
        Object {
          "height": 48,
          "minWidth": 48,
        },
        Object {
          "height": 96,
        },
        undefined,
        Object {
          "backgroundColor": "#FFF",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 8,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
              },
              undefined,
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginHorizontal": 4,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Button Test if button renders properly with size small props 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
      false,
      false,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderStyle": "solid",
        },
        Object {
          "height": 32,
          "minWidth": 32,
        },
        Object {
          "height": 64,
        },
        undefined,
        Object {
          "backgroundColor": "#FFF",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 8,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
              },
              undefined,
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginHorizontal": 4,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Button Test if button renders properly with text style and size medium props 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
      false,
      false,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderStyle": "solid",
        },
        Object {
          "height": 48,
          "minWidth": 48,
        },
        Object {
          "height": 96,
        },
        undefined,
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
            },
            undefined,
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 4,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Button Test if button renders properly with text style and size small props 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
      false,
      false,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderStyle": "solid",
        },
        Object {
          "height": 32,
          "minWidth": 32,
        },
        Object {
          "height": 64,
        },
        undefined,
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
            },
            undefined,
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 4,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
